src/components/Banner/Banner.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useRef, forwardRef, useMemo } from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import useMergeRefs from "../../hooks/useMergeRefs";
+import { IMAGE_POSITIONS } from "./BannerConstants";
+import "./Banner.scss";
+
+const PRESERVE_VALUE = value => value;
+
+const Banner = forwardRef(
+  (
+    { className, imageAlt, imageSrc, renderTitle, renderSubtitle, title, subtitle, imageClassName, imagePosition },
+    ref
+  ) => {
+    const componentRef = useRef(null);
+    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
+
+    const renderedTitle = useMemo(() => {
+      const computedTitle = renderTitle(title);
+      if (!computedTitle) return null;
+      return <h1 className="banner--title">{computedTitle}</h1>;
+    }, [title, renderTitle]);
+
+    const renderedSubtitle = useMemo(() => {
+      const computedSubtitle = renderSubtitle(subtitle);
+      if (!computedSubtitle) return null;
+      return <h2 className="banner--subtitle">{computedSubtitle}</h2>;
+    }, [subtitle, renderSubtitle]);
+
+    const renderImage = useMemo(() => {
+      if (!imageSrc) return null;
+      return <img src={imageSrc} alt={imageAlt} className={cx("banner--image", imageClassName)} />;
+    }, [imageAlt, imageSrc, imageClassName]);
+
+    return (
+      <aside ref={mergedRef} className={cx("banner", className, `image-position__${imagePosition}`)} role="banner">
```

Comment:
role banner is not what we think it is - very strange
https://www.digitala11y.com/banner-role/

because are using aside i don't think we need a role here (please remove it) i'll ask our a11y advisors if anything else is missing

Reply:
Done